lib/secure_headers.rb

```diff
@@ -45,181 +100,158 @@ def configure &block
   class << self
     def append_features(base)
       base.module_eval do
-        extend ClassMethods
         include InstanceMethods
       end
     end
 
-    def header_hash(options = nil)
+    # Strips out headers not applicable to this request
+    def header_hash_for(request)
+      unless request.scheme == HTTPS
+        secure_headers_request_config(request)[SecureHeaders::StrictTransportSecurity::CONFIG_KEY] = SecureHeaders::OPT_OUT
+        secure_headers_request_config(request)[SecureHeaders::PublicKeyPins::CONFIG_KEY] = SecureHeaders::OPT_OUT
+      end
+
       ALL_HEADER_CLASSES.inject({}) do |memo, klass|
```

Comment:
Using `each_with_object` instead of `inject` would save you a line on [L140](https://github.com/twitter/secureheaders/pull/181/files#diff-e7fcd10663004caed933b95e0ff2e643L65).